lib/common/infrastructure/stacktrace_stream_manager.dart

```diff
@@ -3,21 +3,50 @@ import 'dart:async';
 import 'package:hive/hive.dart';
 import 'package:injectable/injectable.dart';
 
+const _stacktraceBoxName = 'stacktrace';
+
 @lazySingleton
 class StacktraceManager {
   bool isEnabled = false;
-  static const String stacktraceBoxName = 'stacktrace';
+
   String _errorTrace = '';
-  StreamController<String> _stacktraceStreamController =
-      StreamController<String>.broadcast();
+  String _stacktrace = '';
 
-  StreamController<String> _errorStreamController =
-      StreamController<String>.broadcast();
+  var _stacktraceStreamController = StreamController<String>.broadcast();
+  var _errorStreamController = StreamController<String>.broadcast();
 
   Stream<String> get stacktraceStream => _stacktraceStreamController.stream;
 
   Stream<String> get errorStream => _errorStreamController.stream;
 
+  /// Get the stacktrace
+  String get stacktrace {
+    if (!_isBoxOpen()) return '';
+    var box = Hive.box(_stacktraceBoxName);
+    final _stacktrace = box.get(_stacktraceBoxName) ?? '';
+    return _stacktrace;
+  }
+
+  /// Get the error trace
+  String get errorTrace => _errorTrace;
+
+  /// Add new trace to the stacktrace
+  void addTrace(String stepDescription) {
+    if (!isEnabled || !_isBoxOpen()) return;
+
+    // write string in an encrypted Hive box
+    final box = Hive.box(_stacktraceBoxName);
+
+    _stacktrace += stepDescription + '\n***\n***';
```

Comment:
Instead of reading+decrypting full stacktrace from Hive we're storing it in memory and writing as it changes.